packages/metal-quartz-collapse/src/MetalQuartzCollapse.js

```diff
@@ -0,0 +1,273 @@
+'use strict';
+
+import anim from 'metal-anim';
+import core from 'metal';
+import dom from 'metal-dom';
+import { EventHandler } from 'metal-events';
+import State from 'metal-state';
+
+/**
+ * Collapse Metal Quartz component.
+ */
+class MetalQuartzCollapse extends State {
+	/**
+	 * @inheritDoc
+	 */
+	constructor(opt_config) {
+		super(opt_config);
+
+		this.eventHandler_ = new EventHandler();
+		this.transitionEventHandler_ = new EventHandler();
+
+		this.supportsTransitionEnd = this.getTransitionEndEvent_();
+		this.transitionEnd = this.getTransitionEndEvent_() || 'transitionend';
+
+		this.on('headersChanged', this.syncHeaders);
+		this.syncHeaders();
+
+		this.on('collapsedChanged', this.syncCollapsed);
+
+		if (this.content) {
+			this.collapsed ? this.close() : this.open();
+		}
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	disposeInternal() {
+		super.disposeInternal();
+		this.eventHandler_.removeAllListeners();
+		this.transitionEventHandler_.removeAllListeners();
+	}
+
+	/**
+	 * Animates close when `collapsed` is true
+	 */
+	animateClose() {
+		this.content.transitionType = 0;
+
+		this.content.setAttribute('style', 'height: ' + this.content.firstElementChild.offsetHeight + 'px;');
+		dom.addClasses(this.content, this.transitionClass);
+		dom.removeClasses(this.content, this.closedClass);
+		dom.removeClasses(this.content, this.openClass);
+
+		this.content.offsetHeight;
+		this.content.style.removeProperty('height');
+
+		this.shimUnsupportedTransition_(this.content);
+	}
+
+	/**
+	 * Animates open when `collapsed` is false
+	 */
+	animateOpen() {
+		this.content.transitionType = 1;
+
+		dom.removeClasses(this.content, this.closedClass);
+		dom.addClasses(this.content, this.transitionClass);
+		this.content.setAttribute('style', 'height: ' + this.content.firstElementChild.offsetHeight + 'px;');
```

Comment:
I can extract the height thing to its own method.

I'm not sure how to integrate template literals here, this component doesn't have a template.